core/src/main/java/org/jdbi/v3/core/result/ResultIterable.java

```diff
@@ -89,27 +89,79 @@ default void forEach(Consumer<? super T> action) {
     }
 
     /**
-     * Get the only row in the result set.
-     * @throws IllegalStateException if zero or multiple rows are returned
-     * @return the object mapped from the singular row in the results
+     * Returns the only row in the result set. Returns {@code null} if the row itself is
+     * {@code null}.
+     * @throws IllegalStateException if the result set contains zero or multiple rows
+     * @return the only row in the result set.
      */
-    default T findOnly() {
+    default T one() {
```

Comment:
This could be a simple `findOne().orElseThrow(() -> new ...)`

Reply:
This is an important distinction and I'm glad you called it out.

`one()` and `first()` can return null, if the row itself maps to null. Both throw if there are no rows.

`findOne()` and `findFirst()`, because they return `Optional<T>`, conflate "no rows" with "row that contained null" into the single result value `Optional.empty()`. So with the `find_()` methods there's no way to tell the difference.

Reply:
I added clarifications to the developer guide in the latest commit.

Reply:
Ah, indeed!